core/services/workflows/store/store_db.go

```diff
@@ -0,0 +1,331 @@
+	return d.transact(ctx, func(db *DBStore) error {
+		wex := &workflowExecutionRow{
+			ID:         state.ExecutionID,
+			WorkflowID: state.WorkflowID,
+			Status:     state.Status,
+		}
+		err := db.insertWorkflowExecution(ctx, wex)
+		if err != nil {
+			return fmt.Errorf("could not insert workflow execution %s: %w", state.ExecutionID, err)
+		}
+
+		ws := []workflowStepRow{}
+		for _, step := range state.Steps {
+			step, err := stateToStep(step)
+			if err != nil {
+				return err
+			}
+			ws = append(ws, step)
+		}
+		if len(ws) > 0 {
+			return db.upsertSteps(ctx, ws)
+		}
+		return nil
+	})
+}
+
+func (d *DBStore) upsertSteps(ctx context.Context, steps []workflowStepRow) error {
+	sql := `
+	INSERT INTO
+	workflow_steps(workflow_execution_id, ref, status, inputs, output_err, output_value)
+	VALUES (:workflow_execution_id, :ref, :status, :inputs, :output_err, :output_value)
+	ON CONFLICT ON CONSTRAINT uniq_workflow_execution_id_ref
+	DO UPDATE SET
+		workflow_execution_id = EXCLUDED.workflow_execution_id,
+		ref = EXCLUDED.ref,
+		status = EXCLUDED.status,
+		inputs = EXCLUDED.inputs,
+		output_err = EXCLUDED.output_err,
+		output_value = EXCLUDED.output_value;
+	`
+	stmt, args, err := sqlx.Named(sql, steps)
+	if err != nil {
+		return err
+	}
+	stmt = d.db.Rebind(stmt)
+	_, err = d.db.ExecContext(ctx, stmt, args...)
+	return err
+}
+
+func (d *DBStore) insertWorkflowExecution(ctx context.Context, execution *workflowExecutionRow) error {
+	sql := `
+	INSERT INTO
+	workflow_executions(id, workflow_id, status)
+	VALUES ($1, $2, $3)
+	`
+	_, err := d.db.ExecContext(ctx, sql, execution.ID, execution.WorkflowID, execution.Status)
+	return err
+}
+
+func (d *DBStore) transact(ctx context.Context, fn func(*DBStore) error) error {
+	return sqlutil.Transact(
+		ctx,
+		func(ds sqlutil.DataSource) *DBStore {
+			return &DBStore{db: ds}
+		},
+		d.db,
+		nil,
+		fn,
+	)
+}
+
+func (d *DBStore) GetUnfinished(ctx context.Context) ([]ExecutionState, error) {
+	sql := `
+	SELECT
+		workflow_steps.workflow_execution_id AS ws_workflow_execution_id,
+		workflow_steps.ref AS ws_ref,
+		workflow_steps.status AS ws_status,
+		workflow_steps.inputs AS ws_inputs,
+		workflow_steps.output_err AS ws_output_err,
+		workflow_steps.output_value AS ws_output_value,
+		workflow_executions.id AS we_id,
+		workflow_executions.workflow_id AS we_workflow_id,
+		workflow_executions.status AS we_status
+	FROM workflow_executions
+	JOIN workflow_steps
+	ON  workflow_steps.workflow_execution_id = workflow_executions.id
+	WHERE workflow_executions.status = 'started'
+	`
+	joinRecords := []struct {
+		WSWorkflowExecutionID string  `db:"ws_workflow_execution_id"`
+		WSRef                 string  `db:"ws_ref"`
+		WSStatus              string  `db:"ws_status"`
+		WSInputs              []byte  `db:"ws_inputs"`
+		WSOutputErr           *string `db:"ws_output_err"`
+		WSOutputValue         []byte  `db:"ws_output_value"`
+		WEID                  string  `db:"we_id"`
```

Comment:
```suggestion
		WEID                  string  `db:"we_id"`
```

This field is yelling at me 😨

Reply:
Yeah, this is a weird Go idiom, abbrevations are supposed to be capitalized so WorkflowExecution -> WE and identifier -> ID. Hence WEID.

Do you have an alternative suggestion? I could go for an expanded WorkflowExecutionID, WorkflowStepWorkflowExecutionID but that felt more verbose for a throwaway struct